crates/bevy_ui/src/layout/ui_surface.rs

```diff
@@ -143,76 +181,150 @@ impl UiSurface {
         }
     }
 
+    /// Removes camera association to root node
+    /// Shorthand for calling `replace_camera_association(root_node_entity, None)`
+    fn mark_root_node_as_orphaned(&mut self, root_node_entity: Entity) {
+        self.replace_camera_association(root_node_entity, None);
+    }
+
+    /// Reassigns or removes a root node's associated camera entity
+    /// `Some(camera_entity)` - Updates camera association to root node
+    /// `None` - Removes camera association to root node
+    /// Does not check to see if they are the same before performing operations
+    fn replace_camera_association(
+        &mut self,
+        root_node_entity: Entity,
+        new_camera_entity_option: Option<Entity>,
+    ) {
+        if let Some(root_node_data) = self.root_node_data.get_mut(&root_node_entity) {
+            // Clear existing camera association, if any
+            if let Some(old_camera_entity) = root_node_data.camera_entity.take() {
+                let prev_camera_root_nodes = self.camera_root_nodes.get_mut(&old_camera_entity);
+                if let Some(prev_camera_root_nodes) = prev_camera_root_nodes {
+                    prev_camera_root_nodes.remove(&root_node_entity);
+                }
+            }
+
+            // Establish new camera association, if provided
+            if let Some(camera_entity) = new_camera_entity_option {
+                root_node_data.camera_entity.replace(camera_entity);
+                self.camera_root_nodes
+                    .entry(camera_entity)
+                    .or_default()
+                    .insert(root_node_entity);
+            }
+        }
+    }
+
+    /// Creates or updates a root node
+    fn create_or_update_root_node_data(
+        &mut self,
+        root_node_entity: Entity,
+        camera_entity: Entity,
+    ) -> &mut RootNodeData {
+        let user_root_node = *self.entity_to_taffy.get(&root_node_entity).expect("create_or_update_root_node_data called before root_node_entity was added to taffy tree or was previously removed");
```

Comment:
This feels weird coming back to, but if memory is serving today, [it's related to the non-atomic nature of how the layout module (super) could call these methods.](https://github.com/bevyengine/bevy/pull/16362/files#r1838700153). Theoretically, the node should be upserted when added well before we call `set_camera_children`. 

(note how `set_camera_children` is `pub`, not `pub(super)`...)